digiroad2-viite/src/main/scala/fi/liikennevirasto/viite/process/ProjectDeltaCalculator.scala

```diff
@@ -107,8 +111,9 @@ object ProjectDeltaCalculator {
   }
 
   def pair(roadAddress: Seq[RoadAddress], projectLink: Seq[ProjectLink]): Seq[(RoadAddress,ProjectLink)] = {
+    val df = new DecimalFormat("#.###")
     roadAddress.foldLeft(List.empty[(RoadAddress,ProjectLink)]) { case (p, a) =>
-      projectLink.find(b => a.linkId == b.linkId) match {
+      projectLink.find(b => a.linkId == b.linkId && b.endMValue == df.format(a.endMValue - a.startMValue).replace(",", ".").toDouble ) match {
```

Comment:
comparison between an endMValue of b and then segment length of a doesn't make sense.
Double comparison with == is no good. Precision should be used instead.
fi.liikennevirasto.viite.MaxAllowedMValueError

for example:
Math.abs(a.endMValue - b.startMValue) <= MaxAllowedMValueError

A new user story, 719, will address this pairin issue.

Reply:
The comparison between the endMValue of b and the length of a is to compare the segments when the link id have more than one address on road_address table. We cannot assume that values on project_link, because the mValues are not the same in Project_Link and Road_address table.

Reply:
@tvenhola in 
Math.abs(a.endMValue - b.startMValue) <= MaxAllowedMValueError should be
(Math.abs(a.endMValue - b.startMValue) - MaxAllowedMValueError )<= MaxDistanceDiffAllowed
;with  MaxAllowedMValueError  =>  0.001; MaxDistanceDiffAllowed => 1.0 ?